pkg/datapath/linux/config/config.go

```diff
@@ -390,6 +391,17 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["NATIVE_DEV_MAC_BY_IFINDEX(IFINDEX)"] = macByIfIndexMacro
 		cDefinesMap["IS_L3_DEV(ifindex)"] = isL3DevMacro
 	}
+
+	// Enabling NetfilterCompatibleMode increases BPF instruction memory size, and may
+	// cause issues in kernels have lower memory. Hence enabling, this feature
+	// only in kernels supports higher instruction size.
+	// NetfilterCompatibleMode feature can be enabled on all kernels, once we have support
+	// for adding new tail calls in host device[bpf_host].
+	if option.Config.NetfilterCompatibleMode && probes.NewProbeManager().GetMisc().HaveLargeInsnLimit &&
+		(option.Config.InstallIptRules || iptables.KernelHasNetfilter()) {
```

Comment:
Given `NetfilterCompatibleMode` is false by default, we should issue a warning when `probes.NewProbeManager().GetMisc().HaveLargeInsnLimit` is false and disable the option explicitly. That logic should also live in `daemon_main.go`.

Reply:
Sure, Done.